pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,777 @@
+
+        return NotImplemented
+
+    __rmul__ = __mul__
+
+    def __sub__(self, H):
+        r"""The subtraction operation between a LinearCombination and a LinearCombination/Tensor/Observable."""
+        if isinstance(H, (LinearCombination, Hamiltonian, Tensor, Observable)):
+            return self + (-1 * H)
+        return NotImplemented
+
+    def __iadd__(self, H):
+        r"""The inplace addition operation between a LinearCombination and a LinearCombination/Tensor/Observable."""
+        if isinstance(H, numbers.Number) and H == 0:
+            return self
+
+        if isinstance(H, (LinearCombination, Hamiltonian)):
+            self._coeffs = qml.math.concatenate([self._coeffs, H.coeffs], axis=0)
+            self._ops.extend(H.ops.copy())
+            self.simplify()
+            return self
+
+        if isinstance(H, (Tensor, Observable)):
+            self._coeffs = qml.math.concatenate(
+                [self._coeffs, qml.math.cast_like([1.0], self._coeffs)], axis=0
+            )
+            self._ops.append(H)
+            self.simplify()
+            return self
+
+        return NotImplemented
+
+    def __imul__(self, a):
+        r"""The inplace scalar multiplication operation between a scalar and a LinearCombination."""
+        if isinstance(a, (int, float)):
+            self._coeffs = qml.math.multiply(a, self._coeffs)
+            return self
+
+        return NotImplemented
+
+    def __isub__(self, H):
+        r"""The inplace subtraction operation between a LinearCombination and a LinearCombination/Tensor/Observable."""
+        if isinstance(H, (LinearCombination, Hamiltonian, Tensor, Observable)):
+            self.__iadd__(H.__mul__(-1))
+            return self
+        return NotImplemented
+
+    def queue(self, context=qml.QueuingManager):
+        """Queues a qml.LinearCombination instance"""
+        for o in self.ops:
+            context.remove(o)
+        context.append(self)
+        return self
+
+    def map_wires(self, wire_map: dict):
+        """Returns a copy of the current LinearCombination with its wires changed according to the given
+        wire map.
+
+        Args:
+            wire_map (dict): dictionary containing the old wires as keys and the new wires as values
+
+        Returns:
+            .LinearCombination: new LinearCombination
+        """
+        cls = self.__class__
+        new_op = cls.__new__(cls)
+        new_op.data = copy(self.data)
+        new_op._wires = Wires(  # pylint: disable=protected-access
+            [wire_map.get(wire, wire) for wire in self.wires]
+        )
+        new_op._ops = [  # pylint: disable=protected-access
+            op.map_wires(wire_map) for op in self.ops
+        ]
+        for attr, value in vars(self).items():
+            if attr not in {"data", "_wires", "_ops"}:
+                setattr(new_op, attr, value)
+        new_op.hyperparameters["ops"] = new_op._ops  # pylint: disable=protected-access
+        return new_op
```

Comment:
Could we just map all the component ops and then construct a new one, instead of the copying and setting private attributes?

Reply:
Yes I think that makes sense! Not sure what is going on here 😅